src/DssDirectDepositHub.t.sol

```diff
@@ -0,0 +1,863 @@
+        // Make sure unwind calls accrued
+        assertTrue(d3mTestPool.accrued());
+    }
+
+    function testFail_no_reap_mcd_caged() public {
+        _windSystem();
+        // interest is determined by the difference in gem balance to dai debt
+        // by giving extra gems to the Join we simulate interest
+        _giveTokens(TokenLike(address(testGem)), 10 * WAD);
+        testGem.transfer(address(d3mTestPool), 10 * WAD);
+
+        // MCD shutdowns
+        end.cage();
+        end.cage(ilk);
+
+        directDepositHub.reap(ilk);
+    }
+
+    function testFail_no_reap_pool_caged() public {
+        _windSystem();
+        // interest is determined by the difference in gem balance to dai debt
+        // by giving extra gems to the Join we simulate interest
+        _giveTokens(TokenLike(address(testGem)), 10 * WAD);
+        testGem.transfer(address(d3mTestPool), 10 * WAD);
+
+        // module caged
+        directDepositHub.cage(ilk);
+
+        directDepositHub.reap(ilk);
+    }
+
+    function test_exit() public {
+        _windSystem();
+        // Vat is caged for global settlement
+        vat.cage();
+
+        // Simulate DAI holder gets some gems from GS
+        vat.grab(
+            ilk,
+            address(d3mTestPool),
+            address(this),
+            address(this),
+            -int256(50 * WAD),
+            -int256(0)
+        );
+
+        uint256 prevBalance = testGem.balanceOf(address(this));
+
+        // User can exit and get the aDAI
+        directDepositHub.exit(ilk, address(this), 50 * WAD);
+        assertEq(testGem.balanceOf(address(this)), prevBalance + 50 * WAD);
+    }
+
+    function test_cage_pool() public {
+        (, , , , uint256 tic) = directDepositHub.ilks(ilk);
+        assertEq(tic, 0);
+
+        directDepositHub.cage(ilk);
+
+        (, , , , tic) = directDepositHub.ilks(ilk);
+        assertEq(tic, block.timestamp);
+    }
+
+    function testFail_cage_pool_no_auth() public {
+        directDepositHub.deny(address(this));
+        directDepositHub.cage(ilk);
+    }
+
+    function test_cull() public {
+        _windSystem();
+        directDepositHub.cage(ilk);
+
+        (uint256 pink, uint256 part) = vat.urns(ilk, address(d3mTestPool));
+        assertEq(pink, 50 * WAD);
+        assertEq(part, 50 * WAD);
+        uint256 gemBefore = vat.gem(ilk, address(d3mTestPool));
+        assertEq(gemBefore, 0);
+        uint256 sinBefore = vat.sin(vow);
+
+        directDepositHub.cull(ilk);
+
+        (uint256 ink, uint256 art) = vat.urns(ilk, address(d3mTestPool));
+        assertEq(ink, 0);
+        assertEq(art, 0);
+        uint256 gemAfter = vat.gem(ilk, address(d3mTestPool));
+        assertEq(gemAfter, 50 * WAD);
+        assertEq(sinBefore + 50 * RAD, vat.sin(vow));
+        (, , , uint256 culled, ) = directDepositHub.ilks(ilk);
+        assertEq(culled, 1);
+    }
+
+    function test_cull_debt_paid_back() public {
```

Comment:
add additional test cases for debt paid back where:
- wind/unwind
- unwind to 0
- wind to DC

Reply:
will handle these here https://github.com/makerdao/dss-direct-deposit/pull/70